Xamarin.Essentials/Screenshot/Screenshot.macos.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+using AppKit;
+using CoreGraphics;
+using Foundation;
+using ImageIO;
+using MobileCoreServices;
+
+namespace Xamarin.Essentials
+{
+    public static partial class Screenshot
+    {
+        internal static bool PlatformIsCaptureSupported => true;
+
+        const string appServicesPath =
+    "/System/Library/Frameworks/ApplicationServices.framework/Versions/Current/ApplicationServices";
+
+        // https://developer.apple.com/documentation/coregraphics/1454595-cgdisplaycreateimage
+        [DllImport(appServicesPath, EntryPoint = "CGDisplayCreateImage")]
+        static extern /* CGImageRef */ IntPtr CGDisplayCreateImage(int displayId);
```

Comment:
It might be worth adding a small helper class in the Platform file so we don't mix functionality with implementation: https://github.com/xamarin/Essentials/blob/main/Xamarin.Essentials/Platform/Platform.macos.cs#L23

You can probably also return the CGImage from there as well so this file only works with concrete types.